src/meta/src/barrier/mod.rs

```diff
@@ -782,15 +593,18 @@ impl GlobalBarrierManager {
             checkpoint,
             span,
         } = self.scheduled_barriers.pop_or_default().await;
-        self.checkpoint_control.pre_resolve(&command);
-        let info = self
+
+        let all_nodes = self
```

Comment:
Is it possible to also maintain a snapshot and gets notify on update instead of listing active compute node for every barrier?

Reply:
Yes, I was plan to do that but found that there may be situations where the notification is not processed in a timely manner, resulting in some gaps when running the command. So I didn't change it in this PR and left a `TODO` in the function `resolve_worker_nodes`. NTFS.

Reply:
For this `handle_new_barrier`, it's called right in the barrier manager `select!` worker loop, and in `handle_new_barrier`, before this `list_active_streaming_compute_nodes`, there is no actually blocking await point (`self.scheduled_barriers.pop_or_default()` is expected to be non-blocking). So if we receive notification in the `select!` and update the compute node info, it should be as timely as the current implementation. But it's fine to implement it in another PR.

Reply:
Actually I'm worry about the situation that when a new computer node joins right after bootstrap recovery (snapshot resolved) and before starting to subscribe compute node changes. This will result in its information not being included in the snapshot. One way to cover the case is to forward subscription before recovery and allow duplicate updates for the info. I will try it in another PR.